File: Week 2/Python Codes/RomanNumberToInteger.py
def roman_to_integer(string):
    my_dict = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
    roman = 0
    for i in range(len(string)):
        if i != len(string) - 1 and my_dict[string[i]] < my_dict[string[i+1]]:
            roman -= my_dict[string[i]]
        else:
            roman += my_dict[string[i]]
    return roman
    
if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        roman = input()
        integer = roman_to_integer(roman)   
        print(integer)

File: Week 2/C++ Codes/RomanNumberToInteger.cpp
#include<bits/stdc++.h>
using namespace std;

int value(char r){
    switch (r) {
        case 'I': return 1;
        case 'V': return 5;
        case 'X': return 10;
        case 'L': return 50;
        case 'C': return 100;
        case 'D': return 500;
        case 'M': return 1000;
    }
}

int romanToInteger(string S) {
    int roman = 0;
    for(int i=0; i<S.length(); i++) {
        if (i != S.length() - 1 and value(S[i]) < value(S[i+1]))
            roman -= value(S[i]);
        else
            roman += value(S[i]);        
    }
    return roman;
}

int main() {
    int t;
    cin>>t;
    while(t--) {
        string S;
        cin>>S;
        cout<<romanToInteger(S)<<endl;
    }
    return 0;
}